psi4/driver/p4util/util.py

```diff
@@ -138,33 +150,39 @@ def set_memory(inputval):
         units = ''
     # Handle memory given as a string
     elif isinstance(inputval, str):
-        memory_string = re.compile(r'^\s*?(\d*\.?\d+)\s+([KMGTPBE]i?B)$', re.IGNORECASE)
+        memory_string = re.compile(r'^\s*(\d*\.?\d+)\s*([KMGTPBE]i?B)\s*$', re.IGNORECASE)
         matchobj = re.search(memory_string, inputval)
         if matchobj:
             val = float(matchobj.group(1))
             units = matchobj.group(2)
         else:
-            raise ValidationError("""Invalid memory specification '{}'.""".format(inputval))
+            raise ValidationError("""Invalid memory specification: {}. Try 5e9 or '5 gb'.""".format(repr(inputval)))
     else:
-        raise TypeError("""Invalid type {0} for argument '{1}'. Supported argument types: 'int', 'float', or 'str'.""".format(type(inputval), inputval))
-        
+        raise ValidationError("""Invalid type {} in memory specification: {}. Try 5e9 or '5 gb'.""".format(
+            type(inputval), repr(inputval)))
+
     # Units decimal or binary?
     multiplier = 1000
     if "i" in units.lower():
         multiplier = 1024
         units = units.lower().replace("i", "").upper()
 
-    # Build conversion factor, convert units        
+    # Build conversion factor, convert units
     unit_list = ["", "KB", "MB", "GB", "TB", "PB", "EB"]
     mult = 1
     for unit in unit_list:
         if units.upper() == unit:
             break
         mult *= multiplier
-            
-    memory_amount = int(val * mult)     
-    
-    core.set_memory(memory_amount)
+
+    memory_amount = int(val * mult)
+
+    min_mem_allowed = 524288000
+    if memory_amount < min_mem_allowed:
+        raise ValidationError("""Please, sir, I want some memory. ({} < {} b)""".format(memory_amount, min_mem_allowed))
```

Comment:
Psi4 has got to be the sassiest qc package out there.  Can we add that to the official tag line? I can see it now... "Psi4: The Sassiest, Freest, and Open-Sourciest Quantum Chemistry Package"